File: samples/ObjC Automatic Sample/watchkitapp Extension/RegistrationController.m
/* IBM Confidential
 * OCO Source Materials
 * 5725E28, 5725S01, 5725I03
 * © Copyright IBM Corp. 2017, 2017
 *
 * The source code for this program is not published or otherwise
 * divested of its trade secrets, irrespective of what has been
 * deposited with the U.S. Copyright Office.
 */

#import "RegistrationController.h"
#import <IBMMobilePushWatch/IBMMobilePushWatch.h>

@interface  RegistrationController()
@property id observer;
@end

@implementation RegistrationController

- (void)willDisappear
{
    [super willDisappear];
    [NSNotificationCenter.defaultCenter removeObserver:self.observer];
}

- (void)willActivate
{
    [super willActivate];
    [self updateRegistrationLabels];
    self.observer = [NSNotificationCenter.defaultCenter addObserverForName:RegisteredNotification object:nil queue:NSOperationQueue.mainQueue usingBlock:^(NSNotification*notification){
        [self updateRegistrationLabels];
    }];
}

-(void)updateRegistrationLabels
{
    if(MCERegistrationDetails.sharedInstance.mceRegistered)
    {
        [self.userIdLabel setText: MCERegistrationDetails.sharedInstance.userId];
        [self.channelIdLabel setText: MCERegistrationDetails.sharedInstance.channelId];
        [self.appKeyLabel setText: MCERegistrationDetails.sharedInstance.appKey];
    }
}

@end

File: samples/Swift Automatic Sample/SampleWatch Extension/RegistrationController.swift
/*
 * Licensed Materials - Property of IBM
 *
 * 5725E28, 5725I03
 *
 * © Copyright IBM Corp. 2017, 2018
 * US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 */

import Foundation
import WatchKit

class RegistrationController: WKInterfaceController
{
    @IBOutlet weak var userIdLabel: WKInterfaceLabel?
    @IBOutlet weak var channelIdLabel: WKInterfaceLabel?
    @IBOutlet weak var appKeyLabel: WKInterfaceLabel?
    var observer: NSObjectProtocol?
    
    override func willDisappear() {
        super.willDisappear()
        if let observer = observer
        {
            NotificationCenter.default.removeObserver(observer)
        }
    }
    
    override func willActivate() {
        super.willActivate()
        updateRegistrationLabels()
        observer = NotificationCenter.default.addObserver(forName:   MCENotificationName.MCERegistered.rawValue, object: nil, queue: OperationQueue.main, using: { (note) in
            self.updateRegistrationLabels()
        })
    }
    
    func updateRegistrationLabels()
    {
        guard let userIdLabel = userIdLabel else
        {
            return
        }
        guard let channelIdLabel = channelIdLabel else
        {
            return
        }
        guard let appKeyLabel = appKeyLabel else
        {
            return
        }
        
        if MCERegistrationDetails.shared.mceRegistered
        {
            userIdLabel.setText(MCERegistrationDetails.shared.userId)
            channelIdLabel.setText(MCERegistrationDetails.shared.channelId)
            appKeyLabel.setText(MCERegistrationDetails.shared.appKey)
        }        
    }
    
}

